packages/client/lib/service/txpool.ts

```diff
@@ -438,8 +450,23 @@ export class TxPool {
       for (const peer of peers) {
         // This is used to avoid re-sending along pooledTxHashes
         // announcements/re-broadcasts
-        this.addToKnownByPeer(hashes, peer)
-        peer.eth?.send('Transactions', txs)
+        const newHashes = this.addToKnownByPeer(hashes, peer)
+        const newHashesHex = newHashes.map((txHash) => txHash.toString('hex'))
+        const nexTxs = txs.filter((tx) => newHashesHex.includes(tx.hash().toString('hex')))
+        peer.eth?.request('Transactions', nexTxs).catch((e) => {
+          this.markFailedSends(peer, newHashes, e as Error)
+        })
+      }
+    }
+  }
+
+  private markFailedSends(peer: Peer, failedHashes: Buffer[], e: Error): void {
```

Comment:
Is it hard to add a test for this function?  I can dig into it maybe on Monday but feels like we should add coverage for this and the error conditions that lead to it.

Reply:
i can test it when i take a break from withdrawals work